### PR TITLE
Ruby: Update link texts to clearly describe it's content

### PR DESCRIPTION
**Pull request by davidumoru**
## Because
Update the link texts to clearly describe the content it leads to, following the accessibility guidelines outlined in the issue.

## This PR
- Update link texts to the sections [Nested Collections](https://www.theodinproject.com/lessons/ruby-nested-collections) && [Project: Caesar Cipher](https://www.theodinproject.com/lessons/ruby-caesar-cipher) as per the given [spreadsheet](https://docs.google.com/spreadsheets/d/1rM9tjPIFZbKoMfd6OlLnjOhddBe2bivllY8_Bg5aGqY/edit?usp=sharing)

## Issue
Related to https://github.com/TheOdinProject/curriculum/issues/28290

## Additional Information
none

## Pull Request Requirements
-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: ruby/basic_ruby/nested_collections.md [hunk 1]
```diff
@@ -322,9 +322,9 @@ Amazing! We have found a great solution to returning an array that only contains
 ### Assignment
 <div class="lesson-content__panel" markdown="1">
 
-1. To learn more about using a hash with a nested array, read [this article](https://web.archive.org/web/20220525215038/https://learn.co/lessons/nested-hash-iteration) from Learn.co.
-2. To learn more about using an array with a nested hash, read [the answer to this post](https://stackoverflow.com/questions/50529389/manipulating-output-from-an-array-of-nested-hashes-in-ruby) on Stack Overflow.
-3. Complete the [nested collections](https://github.com/TheOdinProject/ruby-exercises/tree/main/ruby_basics) exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned.
+1. Read more about [using a hash with a nested array](https://web.archive.org/web/20220525215038/https://learn.co/lessons/nested-hash-iteration).
+2. This Stack Overflow answer explains more about [using arrays with a nested hash](https://stackoverflow.com/questions/50529389/manipulating-output-from-an-array-of-nested-hashes-in-ruby).
+3. Complete the nested collections exercises
```
Comment (MaoShizhong) [inline ruby/basic_ruby/nested_collections.md:327]: The text to replace in the spreadsheet was only the first bit of the whole sentence, not the whole sentence itself - easily missed!
```suggestion
3. Complete the nested collections exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned.
```

**Comment by davidumoru** [comment]
Hi @MaoShizhong, let me know if you would like me to fix the linting errors :)

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
One fix detailed below.

On top of this, would you be comfortable fixing the linting issues in both files? We have [instructions for linting and fixing locally](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md) you can follow. The lesson file should be fixable in 1 or 2 runs of the lesson fix script, and the project file should be fixable with 1 run of the project fix script.

If you're not comfortable doing this, let me know - we can merge once the link fix detailed below is pushed. If you are happy to fix those lint errors, awesome!

File: ruby/basic_ruby/nested_collections.md (hunk 1 above)
Comment (davidumoru) [reply ruby/basic_ruby/nested_collections.md]: Noted! I have updated the text and I fixed the lining issues too

File: ruby/basic_ruby/nested_collections.md [hunk 2]
```diff
@@ -320,32 +330,34 @@ vehicles.filter_map { |name, data| name if data[:year] >= 2020 }
...
 ### Assignment
+
 <div class="lesson-content__panel" markdown="1">
 
-1. To learn more about using a hash with a nested array, read [this article](https://web.archive.org/web/20220525215038/https://learn.co/lessons/nested-hash-iteration) from Learn.co.
-2. To learn more about using an array with a nested hash, read [the answer to this post](https://stackoverflow.com/questions/50529389/manipulating-output-from-an-array-of-nested-hashes-in-ruby) on Stack Overflow.
-3. Complete the [nested collections](https://github.com/TheOdinProject/ruby-exercises/tree/main/ruby_basics) exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned.
+1. Read more about [using a hash with a nested array](https://web.archive.org/web/20220525215038/https://learn.co/lessons/nested-hash-iteration).
+1. This Stack Overflow answer explains more about [using arrays with a nested hash](https://stackoverflow.com/questions/50529389/manipulating-output-from-an-array-of-nested-hashes-in-ruby).
+1. Complete the nested collections exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises)
```
Comment (MaoShizhong) [inline ruby/basic_ruby/nested_collections.md:338]: A definite nit, but we should be consistent with punctuation.
```suggestion
1. Complete the nested collections exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises).
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Thanks for fixing the lint errors and link bug!

File: ruby/basic_ruby/nested_collections.md (hunk 2 above)
Comment (davidumoru) [reply ruby/basic_ruby/nested_collections.md]: Done!

**Review by MaoShizhong — APPROVED** [review]
Awesome! Congrats on your first PR here.

**Pull request merged**